src/Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests/PullDeliveryTests.cs

```diff
@@ -0,0 +1,167 @@
+        await runTask;
+    }
+
+    private IHost BuildHost(string eventGridUrl)
+    {
+        IHostBuilder? builder = Host.CreateDefaultBuilder();
+        builder.ConfigureServices(services =>
+        {
+            services.AddSingleton(Channel.CreateUnbounded<TestEvent>());
+            services.AddEventPropagationPublisher(options =>
+            {
+                options.TopicEndpoint = eventGridUrl;
+                options.TopicName = TopicName;
+                options.TopicAccessKey = "noop";
+                options.TopicType = TopicType.Namespace;
+            });
+            services.AddPullDeliverySubscription()
+                .AddTopicSubscription("DummySectionName", options =>
+                {
+                    options.TopicEndpoint = eventGridUrl;
+                    options.TopicName = TopicName;
+                    options.SubscriptionName = SubscriberName;
+                    options.TopicAccessKey = "noop";
+                })
+                .AddDomainEventHandler<TestEvent, TestDomainEventHandler>();
+        });
+
+        return builder.Build();
+    }
+
+    [DomainEvent("com.workleap.sample.testEvent", EventSchema.CloudEvent)]
+    private sealed record TestEvent : IDomainEvent
+    {
+        public int Id { get; set; }
+    }
+
+    private sealed class TestDomainEventHandler(Channel<TestEvent> channel) : IDomainEventHandler<TestEvent>
+    {
+        public async Task HandleDomainEventAsync(TestEvent domainEvent, CancellationToken cancellationToken)
+        {
+            await channel.Writer.WriteAsync(domainEvent, cancellationToken);
+        }
+    }
+
+    private sealed class EventGridEmulatorContext(IContainer container) : IAsyncDisposable
+    {
+        public string Url { get; } = $"http://localhost:{container.GetMappedPublicPort(6500)}/";
+
+        public static async Task<EventGridEmulatorContext> StartAsync(ITestOutputHelper testOutputHelper)
+        {
+            var configurationPath = await WriteConfigurationFile(testOutputHelper);
+            
+            var container = BuildContainer(configurationPath);
+            
+            try
+            {
+                await container.StartAsync();
+            }
+            catch
+            {
+                try
+                {
+                    var logs = await container.GetLogsAsync();
+                    testOutputHelper.WriteLine(logs.Stdout);
+                    testOutputHelper.WriteLine(logs.Stderr);
+                }
+                catch
+                {
+                    // do not hide the original exception
+                }
+
+                throw;
+            }
+
+            return new EventGridEmulatorContext(container);
+        }
+
+        private static IContainer BuildContainer(string configurationPath)
+        {
+            return new ContainerBuilder()
+                .WithImage("workleap/eventgridemulator:0.2.0") // TODO Renovate this?
```

Comment:
regarding the hardocded image version, would this necessary to update through renovate (potentially it may require custom logic for .cs files)